src/Native/System.IO.Compression.Native/pal_zlib.h

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "pal_types.h"
+
+/*
+A structure that holds the input and output values for the zlib functions.
+*/
+struct PAL_ZStream
+{
+    uint8_t* nextIn; // next input byte
+    uint8_t* nextOut; // next output byte should be put there
+
+    uint32_t availIn; // number of bytes available at nextIn
+    uint32_t availOut; // remaining free space at nextOut
```

Comment:
The best way to avoid internal padding and have proper alignment is to put the widest members first, starting with pointers. This ensures that the struct starts aligned and each member is packed. On second look, because you have two 32bit ints next to each other, you may get away with it here (on 63 and 32 bit systems since the pointers will be 64 bit and two 32 bit ints can be packed together to keep alignment). Feel free to won't fix since this should actually be a no-op on our current supported systems (64 and 32 bit) but just something to keep in mind :) sorry about that!